book/continuous_integration.md

```diff
@@ -0,0 +1,171 @@
+        
+    - name: Test with pytest
+      run: |
+        pytest
+```
+
+The first section of this example describes when our workflow should be run. In this case, we're running the CI workflow whenever code is `push`ed to the `master` branch or where any Pull Request is created. In the case of Pull Requests, the results of the CI workflow will be report on the request's page. If any of the workflow stages fail, this can block the merge of these changes onto a more stable branch. Subsequent commits to the source branch will trigger the CI workflow to run again.
+
+Below `jobs`, we're defining what tasks we would like to run when our workflow is triggered. We define what operating system we would like to run our workflow on - the Linux operating system `ubuntu` here. The `matrix` section under `strategy` defines parameters for the workflow. The workflow will be repeated for each combination of parameters supplied here - in this case 4 recent Python versions.
+
+The individual stages of the workflow are defined under `steps`. `steps` typically have an informative name and run code to perform an action. Here `uses: actions/checkout@v3` references [existing code](https://github.com/actions/checkout) that will retrieve the code from our repo. The subsequent `steps` will use this code. The next step provides us with a specific Python version, as specified in the `matrix`. Then we install dependencies/requirements for our code and the `pytest` module. Finally, we run `pytest` to check that our code is working as expected.
+
+This workflow will report whether our test code ran successfully for each of the specified Python versions.
+
+#### Configure GitHub actions to build and deploy documentation
+
+It is important to maintain the documentation relating to your project to ensure contributors and users can understand, maintain and use your product correctly. One basic way of doing this is maintaining markdown files within a GitHub repository. However, there exists multiple tools that can transform these markdown files into HTML content. A popular tool for building and deploying HTML documentation is [Sphinx](https://www.sphinx-doc.org/en/master/). Here are two examples of repositories that use sphinx to build its documentation:
+
+* [Quality assurance of code for analysis and research (this book)](https://github.com/best-practice-and-impact/qa-of-code-guidance/blob/main/.github/workflows/book.yaml)
+* [govcookiecutter](https://github.com/best-practice-and-impact/govcookiecutter/blob/main/.github/workflows/govcookiecutter-deploy-documentation.yml)
+
+### Example GitLab Pipeline
+
+GitLab has an equivalent to GitHub Actions called GitLab Pipelines. The use cases for these are practically the same, with a change in syntax and file structure. [This blog](https://www.patricksoftwareblog.com/setting-up-gitlab-ci-for-a-python-application/) provides a simple GitLab pipeline example and detailed description on how to use it. 
```

Comment:
Would steer away from using 'this blog' as hyperlink text for accessibility reasons.

Reply:
I've chosen to name the blog for accessibility purposes.